controllers/clustermanagementaddon_controller.go

```diff
@@ -197,55 +207,84 @@ func (c *ClusterManagementAddonReconciler) deployProxyServer(config *proxyv1alph
 		newProxyServerRole(config),
 		newProxyServerRoleBinding(config),
 	}
+	deployed := false
 	for _, resource := range resources {
-		if err := c.ensure(resource); err != nil {
-			return err
+		created, updated, err := c.ensure(config.Generation, resource)
```

Comment:
nit I might only consider if the resource is modified (created or updated). It seems created, updated can be reduced to 1 var

Reply:
actually i clarified them on purpose, to record events when any resources are modified. this will be helpful for us to find out the timeline when some resources are unexpectedly updated.